src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendHelper.java

```diff
@@ -6,22 +6,41 @@
 import com.contrastsecurity.models.AgentType;
 import com.contrastsecurity.models.Rules;
 import com.contrastsecurity.sdk.ContrastSDK;
+import hudson.ProxyConfiguration;
 import hudson.model.Run;
 import hudson.model.TaskListener;
 import hudson.util.ListBoxModel;
+import jenkins.model.Jenkins;
 import org.apache.commons.lang3.StringUtils;
 
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.EnumSet;
-import java.util.List;
+import java.net.MalformedURLException;
+import java.net.Proxy;
+import java.net.URL;
+import java.util.*;
 
 public class VulnerabilityTrendHelper {
 
     public static ContrastSDK createSDK(String username, String serviceKey, String apiKey, String teamServerUrl) {
-        return new ContrastSDK(username, serviceKey, apiKey, teamServerUrl);
+        ContrastSDK contrastSDK = null;
+        ProxyConfiguration proxyConfig = Jenkins.getInstance().proxy;
+        URL url = null;
+        try {
+            url = new URL(teamServerUrl);
+        } catch (MalformedURLException e) {
+            e.printStackTrace();
+        }
+        if (proxyConfig != null && url != null) {
```

Comment:
We can still do better here. Tough to read this out loud...

`if there is a proxy config we skip this block...`
`if there isn't a contrastSDK, then...`

1) Write some unit tests around `createSDK`. You have several cases to test
`No proxy config`
`ProxyConfig but no proxy`
`Proxy config and a proxy`
`Malformed teamserverUrl`

2) Then come back to this method and refactor it for clarity

Reply:
Done, the function is refactored